js/src/modal.js

```diff
@@ -537,53 +538,57 @@ const Modal = (() => {
    * ------------------------------------------------------------------------
    */
 
-  $(document).on(Event.CLICK_DATA_API, Selector.DATA_TOGGLE, function (event) {
+  EventHandler.on(document, Event.CLICK_DATA_API, Selector.DATA_TOGGLE, function (event) {
     let target
     const selector = Util.getSelectorFromElement(this)
 
     if (selector) {
-      target = $(selector)[0]
+      target = SelectorEngine.findOne(selector)
     }
 
-    const config = $(target).data(DATA_KEY) ?
-      'toggle' : $.extend({}, $(target).data(), $(this).data())
+    const config = Data.getData(target, DATA_KEY) ?
+      'toggle' : Util.extend({}, Util.getDataAttributes(target), Util.getDataAttributes(this))
 
     if (this.tagName === 'A' || this.tagName === 'AREA') {
       event.preventDefault()
     }
 
-    const $target = $(target).one(Event.SHOW, (showEvent) => {
-      if (showEvent.isDefaultPrevented()) {
+    EventHandler.one(target, Event.SHOW, (showEvent) => {
+      if (showEvent.defaultPrevented) {
         // only register focus restorer if modal will actually get shown
         return
       }
 
-      $target.one(Event.HIDDEN, () => {
-        if ($(this).is(':visible')) {
+      EventHandler.one(target, Event.HIDDEN, () => {
+        if (Util.isVisible(this)) {
           this.focus()
         }
       })
     })
 
+    const $ = window.$ || window.jQuery
     Modal._jQueryInterface.call($(target), config, this)
```

Comment:
Without a change here, our Modal continue to depend on jQuery

Reply:
I've missed it. I've just changed it.